Sources/Filters/General/ContourTriangulator/helper.js

```diff
@@ -0,0 +1,2290 @@
+    const k = idx++;
+    poly1[i2] = innerPoly[k];
+    edges1[i2] = innerEdges[k];
+    idx *= idx !== m;
+  }
+  edges1[n2 - 1] = -1;
+
+  // Generate poly2
+  const m1 = n * (cuts[0][0] < cuts[1][0]) + cuts[0][0] - cuts[1][0] + 1;
+  const m2 = m1 + m * (cuts[1][1] < cuts[0][1]) + cuts[1][1] - cuts[0][1] + 1;
+
+  // TODO: Check
+  // vtkCCSPoly poly2(m2);
+  const poly2 = [];
+  poly2.length = m2;
+  // vtkCCSPolyEdges edges2(m2);
+  const edges2 = [];
+  edges2.length = m2;
+
+  idx = cuts[1][0];
+  for (let j1 = 0; j1 < m1; j1++) {
+    const k = idx++;
+    poly2[j1] = outerPoly[k];
+    edges2[j1] = outerEdges[k];
+    idx *= idx !== n;
+  }
+  edges2[m1 - 1] = -1;
+
+  idx = cuts[0][1];
+  for (let j2 = m1; j2 < m2; j2++) {
+    const k = idx++;
+    poly2[j2] = innerPoly[k];
+    edges2[j2] = innerEdges[k];
+    idx *= idx !== m;
+  }
+  edges2[m2 - 1] = -1;
+
+  // Replace outerPoly and innerPoly with these new polys
+  polys[outerPolyId] = poly1;
+  polys[innerPolyId] = poly2;
+  polyEdges[outerPolyId] = edges1;
+  polyEdges[innerPolyId] = edges2;
+}
+
+// ---------------------------------------------------
+/**
+ * After the holes have been identified, make cuts between the
+ * outer poly and each hole.  Make two cuts per hole.  The only
+ * strict requirement is that the cut must not intersect any
+ * edges, but it's best to make sure that no really sharp angles
+ * are created.
+ *
+ * @param {Number[][]} polys
+ * @param {vtkPoints} points
+ * @param {Number[][]} polyGroups
+ * @param {Number[]} polyEdges
+ * @param {Vector3} normal
+ */
+export function vtkCCSCutHoleyPolys(
+  polys,
+  points,
+  polyGroups,
+  polyEdges,
+  normal
+) {
+  let cutFailure = 0;
+
+  // Go through all groups and cut out the first inner poly that is
+  // found.  Every time an inner poly is cut out, the groupId counter
+  // is reset because cutting a poly creates a new group.
+  let groupId = 0;
+  while (groupId < polyGroups.length) {
+    const polyGroup = polyGroups[groupId];
+
+    // Only need to make a cut if the group size is greater than 1
+    if (polyGroup.length > 1) {
+      // The first member of the group is the outer poly
+      const outerPolyId = polyGroup[0];
+
+      // The second member of the group is the first inner poly
+      let innerPolyId = polyGroup[1];
+
+      // Sort the group by size, do largest holes first
+      // TODO: Check
+      // std::vector<std::pair<size_t, size_t>> innerBySize(polyGroup.size());
+      let innerBySize = [];
+
+      for (let i = 1; i < polyGroup.length; i++) {
+        // TODO: Check
+        // innerBySize[i].first = polys[polyGroup[i]].size();
+        // innerBySize[i].second = i;
+        innerBySize[i] = [polys[polyGroup[i]].length, i];
+      }
+
+      // TODO: Check
+      // std::sort(innerBySize.begin() + 1, innerBySize.end());
+      // std::reverse(innerBySize.begin() + 1, innerBySize.end());
+      innerBySize = [innerBySize[0], ...innerBySize.splice(1).sort()];
```

Comment:
sort is not working here

Reply:
```
innerBySize = [
        innerBySize[0],
        ...innerBySize.splice(1).sort((a, b) => a[0] - b[0]),
      ];
```

C++ sorts pairs by their first element per default.

Reply:
👍